homeassistant/components/light/tplink.py

```diff
@@ -149,17 +149,15 @@ def update(self):
             if self._supported_features & SUPPORT_RGB_COLOR:
                 self._rgb = hsv_to_rgb(self.smartbulb.hsv)
             if self.smartbulb.has_emeter:
-                self._emeter_params[ATTR_CURRENT_CONSUMPTION] \
-                    = "%.1f W" % self.smartbulb.current_consumption()
+                self._emeter_params[ATTR_CURRENT_POWER_W] \
+                    = self.smartbulb.current_consumption() / 1e3
```

Comment:
I think you should keep the rounding in the string formatting if it makes sense. But please use new style string formatting instead.
```py
'{:.1f}'.format(self.smartbulb.current_consumption() / 1e3)
```